.gitlab-ci.yml

```diff
@@ -972,62 +972,64 @@ deploy_windows_testing-a7:
   script:
     - $S3_CP_CMD --recursive --exclude "*" --include "datadog-agent-7.*.msi" $OMNIBUS_PACKAGE_DIR s3://$WINDOWS_TESTING_S3_BUCKET_A7 --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers full=id=3a6e02b08553fd157ae3fb918945dd1eaae5a1aa818940381ef07a430cf25732
 
-
 #
 # Kitchen Test Common templates
 #
 
 .kitchen_common: &kitchen_common
-  script:
-    - cd $DD_AGENT_TESTING_DIR
-    - rm -rf $CI_PROJECT_DIR/kitchen_logs
-    - rm -rf $DD_AGENT_TESTING_DIR/.kitchen
-    - mkdir $CI_PROJECT_DIR/kitchen_logs
-    - ln -s $CI_PROJECT_DIR/kitchen_logs $DD_AGENT_TESTING_DIR/.kitchen
-    - bash -l tasks/run-test-kitchen.sh
+  <<: *run_when_testkitchen_triggered
+  stage: testkitchen_testing
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/dd-agent-testing:$DATADOG_AGENT_BUILDERS
   artifacts:
     expire_in: 2 weeks
     when: always
     paths:
       - $CI_PROJECT_DIR/kitchen_logs
+  tags: [ "runner:main", "size:large" ]
+  retry: 1
 
-.kitchen_windows_installer_common: &kitchen_windows_installer_common
+.kitchen_a6_common: &kitchen_a6_common
+  variables:
+    PYTHON_RUNTIMES: '2,3'
+    DD_PIPELINE_ID: $CI_PIPELINE_ID-a6
+
+.kitchen_a7_common: &kitchen_a7_common
+  variables:
+    PYTHON_RUNTIMES: '3'
```

Comment:
It's a bit hacky that this makes the test use agent 7. If not easy to remove, at least let's add a comment.

Reply:
I was thinking the same thing when I was refactoring this file. I actually think this variable is not used (only `DD_PIPELINE_ID` seems to be used in the kitchen tests), but I'll test if this can be removed in another PR after this is merged (I don't want to do too many changes in a single PR).